pkg/library/workspace/helper.go

```diff
@@ -0,0 +1,43 @@
+//
+// Copyright (c) 2019-2022 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package workspace
+
+import (
+	"github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+	dw "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+	"github.com/devfile/devworkspace-operator/pkg/config"
+)
+
+// Overwrites the content of the workspace's Template Spec with the workspace config's Template Spec,
+// with the exception of the workspace's projects.
+// If the workspace's Template Spec defined any projects, they are preserved.
+func ApplyDefaultTemplate(workspace *dw.DevWorkspace) {
+	var projectsCopy []v1alpha2.Project
+	if len(workspace.Spec.Template.Projects) > 0 {
+		projectsCopy = make([]v1alpha2.Project, len(workspace.Spec.Template.Projects))
+		copy(projectsCopy, workspace.Spec.Template.Projects)
+	}
+
+	config.Workspace.DefaultTemplate.DeepCopyInto(&workspace.Spec.Template.DevWorkspaceTemplateSpecContent)
+
+	if len(projectsCopy) > 0 {
+		if workspace.Spec.Template.Projects == nil {
+			workspace.Spec.Template.Projects = projectsCopy
+		} else {
+			workspace.Spec.Template.Projects = append(workspace.Spec.Template.Projects, projectsCopy...)
+		}
+	}
+}
```

Comment:
```suggestion
func ApplyDefaultTemplate(workspace *dw.DevWorkspace) {
	if config.Workspace.DefaultTemplate == nil {
		return
	}
	defaultCopy := config.Workspace.DefaultTemplate.DeepCopy()
	originalProjects := workspace.Spec.Template.Projects
	workspace.Spec.Template.DevWorkspaceTemplateSpecContent = *defaultCopy
	for _, project := range originalProjects {
		workspace.Spec.Template.Projects = append(workspace.Spec.Template.Projects, project)
	}
}
```

Reply:
Much cleaner thank you :) I really overcomplicated it there :sweat_smile: And good catch on checking whether the default template is `nil`